langchain4j-google-ai-gemini/src/test/java/dev/langchain4j/model/googleai/GoogleAiGeminiChatModelIT.java

```diff
@@ -703,6 +708,81 @@ void should_count_tokens() {
         assertThat(listOfMsgTokenCount).isGreaterThan(0);
     }
 
+    static class Calculator {
+        @Tool
+        public int add(int a, int b) {
+            System.out.println("ADD " + a + " + " + b);
+            return a + b;
+        }
+
+        @Tool
+        public int subtract(int a, int b) {
+            System.out.println("SUBTRACT " + a + " - " + b);
+            return a - b;
+        }
+
+        @Tool
+        public int multiply(int a, int b) {
+            System.out.println("MULTIPLY " + a + " * " + b);
+            return a * b;
+        }
+
+        @Tool
+        public int divide(int a, int b) {
+            System.out.println("DIVIDE " + a + " / " + b);
+            return a / b;
+        }
+    }
+
+    interface Assistant {
+        @dev.langchain4j.service.SystemMessage(
+            "When asked to evaluate some math expressions, " +
+            "you MUST use the `add`, `substract`, `multiply`, and `divide` tool functions."
+        )
+        String chat(String userMessage);
+    }
+
+    @RetryingTest(3)
+    void should_work_with_tools_with_AiServices() {
+        // given
+        GoogleAiGeminiChatModel gemini = GoogleAiGeminiChatModel.builder()
+            .apiKey(GOOGLE_AI_GEMINI_API_KEY)
+            .modelName("gemini-1.5-pro")
+            .logRequestsAndResponses(true)
+            .build();
+
+        // when
+        Calculator spyCalculator = spy(new Calculator());
+
+        MessageWindowChatMemory chatMemory = MessageWindowChatMemory.withMaxMessages(20);
+        Assistant assistant = AiServices.builder(Assistant.class)
+            .tools(spyCalculator)
+            .chatMemory(chatMemory)
+            .chatLanguageModel(gemini)
+            .build();
+
+        // then
+        String response = "";
+
+        response = assistant.chat("How much is 3 + 4?");
+        assertThat(response).containsIgnoringCase("7");
+        verify(spyCalculator).add(3, 4);
+
+        response = assistant.chat("How much is 18 - 4?");
```

Comment:
minor: not sure having 4 interactions instead of 1 brings a lot of value to the test

Reply:
OK, it's not really critical. Although if you need 2, you could remove 2 and make this test faster and cheaper

Reply:
OK, I'll make just two then.